lib/scope.js

```diff
@@ -1182,37 +1182,98 @@ module.exports = {
-      // sometimes it blocks nodes from clicks. Scroll item to the bottom instead.
-      await handle.evaluate(( elem ) => { elem.scrollIntoView(false); return elem.scrollTop });
-      let set_to = row.value;
-      if ( type === `checkbox` ) { set_to = row.checked; }
-      
-      if ( env_vars.DEBUG && handle ) {
-        let name = await handle.evaluate(( elem ) => {
-          try { elem.nextSibling.style.background = 'Khaki'; }
-          catch(err){ elem.style.background = 'Khaki'; }
-          return elem.getAttribute('for') || elem.getAttribute('name'); 
-        });
-        console.log( `setting elem ${ name }, var ${ row.var } to ${ set_to }` );
-        await scope.page.waitFor( 400 );
+    // If field is disabled, don't set it yet
+    if ( disabled ) { return !disabled; }
+
+    // da's header covers the top of the section containing the fields which means
+    // sometimes it blocks nodes from clicks. Scroll item to the bottom instead.
+    await handle.evaluate(( elem ) => { elem.scrollIntoView(false); return elem.scrollTop });
+    let set_to = row.value;
+    if ( type === `checkbox` ) { set_to = row.checked; }
+
+    if ( env_vars.DEBUG && handle ) {
+      let name = await handle.evaluate(( elem ) => {
+        try { elem.nextSibling.style.background = 'Khaki'; }
+        catch(err){ elem.style.background = 'Khaki'; }
+        return elem.getAttribute('for') || elem.getAttribute('name'); 
+      });
+      await scope.page.waitFor( 400 );
+    }
+
+    // Detect a custom type and get the element of that custom type if it exists
+    let datatype = await handle.evaluate(( elem )=> {
+      // If the element is in a form group* then extract the form group's
+      // datatype value.
+      // * For now the only custom datatype we have is in a form group. In
+      // future we should look into what happens with a `fieldset` for
+      // `choices:`, etc.
+      let custom_datatype = null, matches = null;
+      let group = elem.closest( `.form-group` );
+      if ( group && group.className ) {
+        matches = ` ${ group.className } `.match(/da-field-container-datatype-([^ ]+) /);
+      }
+      if ( matches ) {
+        custom_datatype = matches[1];
       }
 
+      return custom_datatype;
+    });
+
+    // Set field
+    if ( scope.setCustomDatatype[ datatype ] === undefined ) {
+      // If no custom datatype, set a regular field
       await scope.setField[ tag ]( scope, { handle, set_to });
+    } else {
+      // Set da package custom fields
+      let custom_handle = await handle.evaluateHandle(( elem )=> { return elem.closest( `.form-group` ); });
+      await scope.setCustomDatatype[ datatype ]( scope, { handle: custom_handle, set_to });
     }
 
     return !disabled;
   },  // Ends scope.setVariable()
 
+  // Handle setting values for da custom datatypes
+  // E.g. `da-field-container-datatype-BirthDate`
+  // TODO: Make this more easily extensible
+  setCustomDatatype: {
```

Comment:
How can we make `setCustomDatatype` more easily extensible?

Reply:
(at some point in the future)

Reply:
I think this is a good start. This seems extensible, although the exact particulars will depend on what other custom fields we come across.